test/e2e/virtual/apiexport/virtualworkspace_test.go

```diff
@@ -178,6 +162,129 @@ func TestAPIExportVirtualWorkspace(t *testing.T) {
 
 }
 
+func TestAPIExportWithPermissionClaims(t *testing.T) {
+	t.Parallel()
+
+	server := framework.SharedKcpServer(t)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+
+	// Need to Create a Producer w/ APIExport
+	orgClusterName := framework.NewOrganizationFixture(t, server)
+	serviceProviderWorkspace := framework.NewWorkspaceFixture(t, server, orgClusterName, "Universal")
+	serviceProviderSherriffs := framework.NewWorkspaceFixture(t, server, orgClusterName, "Universal")
+	consumerWorkspace := framework.NewWorkspaceFixture(t, server, orgClusterName, "Universal")
+
+	cfg := server.DefaultConfig(t)
+
+	kcpClients, err := clientset.NewClusterForConfig(cfg)
+	require.NoError(t, err, "failed to construct kcp cluster client for server")
+
+	dynamicClients, err := dynamic.NewClusterForConfig(cfg)
+	require.NoError(t, err, "failed to construct dynamic cluster client for server")
+
+	kubeClusterClient, err := kubernetes.NewClusterForConfig(cfg)
+	require.NoError(t, err, "failed to construct kube cluster client for server")
+
+	wildwestClusterClient, err := wildwestclientset.NewClusterForConfig(cfg)
+	require.NoError(t, err, "failed to construct wildwest cluster client for server")
+
+	apifixtures.CreateSheriffsSchemaAndExport(ctx, t, serviceProviderSherriffs, kcpClients, "wild.wild.west", "board the wanderer")
+
+	t.Logf("get the sheriffs api export's generated identity hash")
+	identityHash := ""
+	wait.PollImmediateWithContext(ctx, 100*time.Millisecond, wait.ForeverTestTimeout, func(c context.Context) (done bool, err error) {
```

Comment:
You either need to check the error returned, or switch to `require.Eventually` (I'd do the latter, but up to you)

Reply:
checking err, require was not working.